trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java

```diff
@@ -135,6 +135,17 @@ void loadCatalogs(GravitinoMetaLake metalake) {
         metalake.name(),
         Arrays.toString(catalogNames));
 
+    // Delete those catalogs that have been deleted in Gravitino server
+    catalogConnectors.keySet().stream()
+        .filter((String catalogName) -> !Arrays.asList(catalogNames).contains(catalogName))
+        .forEach(
+            (String catalogName) -> {
+              catalogConnectors.remove(catalogName);
+              catalogInjector.removeCatalogConnector(catalogName);
```

Comment:
You need to remove catalog from Trino first. Following the current sequence, it's possible that a user can retrieve the catalog in Trino, but not in the Gravitino connector.

Reply:
I see.